Assets/MRTK/Core/Utilities/Physics/MixedRealityRaycaster.cs

```diff
@@ -19,96 +25,106 @@ public static bool RaycastSimplePhysicsStep(RayStep step, LayerMask[] prioritize
             return RaycastSimplePhysicsStep(step, step.Length, prioritizedLayerMasks, focusIndividualCompoundCollider, out physicsHit);
         }
 
+        private static readonly ProfilerMarker RaycastSimplePhysicsStepPerfMarker = new ProfilerMarker("[MRTK] MixedRealityRaycaster.RaycastSimplePhysicsStep");
+
         /// <summary>
         /// Simple raycasts each physics <see cref="Microsoft.MixedReality.Toolkit.Physics.RayStep"/> within a specified maximum distance.
         /// </summary>
         /// <returns>Whether or not the raycast hit something.</returns>
         public static bool RaycastSimplePhysicsStep(RayStep step, float maxDistance, LayerMask[] prioritizedLayerMasks, bool focusIndividualCompoundCollider, out RaycastHit physicsHit)
         {
-            Profiler.BeginSample("[MRTK] MixedRealityRaycaster.RaycastSimplePhysicsStep");
-
-            Debug.Assert(maxDistance > 0, "Length must be longer than zero!");
-            Debug.Assert(step.Direction != Vector3.zero, "Invalid step direction!");
-
-            bool result = prioritizedLayerMasks.Length == 1
-                // If there is only one priority, don't prioritize
-                ? UnityEngine.Physics.Raycast(step.Origin, step.Direction, out physicsHit, maxDistance, prioritizedLayerMasks[0])
-                // Raycast across all layers and prioritize
-                : TryGetPrioritizedPhysicsHit(UnityEngine.Physics.RaycastAll(step.Origin, step.Direction, maxDistance, UnityEngine.Physics.AllLayers), prioritizedLayerMasks, focusIndividualCompoundCollider, out physicsHit);
+            using (RaycastSimplePhysicsStepPerfMarker.Auto())
+            {
+                Debug.Assert(maxDistance > 0, "Length must be longer than zero!");
+                Debug.Assert(step.Direction != Vector3.zero, "Invalid step direction!");
 
-            Profiler.EndSample(); // RaycastSimplePhysicsStep
+                bool result = false;
+                if (prioritizedLayerMasks.Length == 1)
+                {
+                    // If there is only one priority, don't prioritize
+                    result = UnityEngine.Physics.Raycast(step.Origin, step.Direction, out physicsHit, maxDistance, prioritizedLayerMasks[0]);
```

Comment:
I wonder if just changing to use `RaycastNonAlloc` here too would be more performant? Even though we only have one layer mask and can let Unity handle the first hit, are we still allocating?

Reply:
good question. if we created a single element array, there would be no need for Unity to allocate the return

Reply:
same could be done for spherecast as well

Reply:
updated.